### PR TITLE
Introduce support for packit builds

### LINKED ISSUE #452: Migrate current COPR snapshot builds to packit
## Describe the bug

We are building and publishing snapshot builds https://copr.fedorainfracloud.org/coprs/mperina/hirte-snapshot/
There are currently 2 problems with COPR:

- It's hard to create and maintain organization in COPR
- If we need to check packaging issues on Fedora/EPEL or different platforms, which are not available on GitHub CI, we need to wait till the change is merged and then check COPR

We may try to switch to https://packit.dev/ which should provide all those missing features.

And also it PackIT approach works, we don't need to handle renaming Hirte to BlueChi in COPR

### PR DESCRIPTION
**Pull request by mwperina**
Introduces ability to build RPMs using packit and run integration tests within testing farm.

Current Github CI to build RPMs and run integrations tests will be remove in follow up PR.

Fixes: https://github.com/containers/bluechi/issues/452

### CODE REVIEW

File: tests/conftest.py [hunk 1]
```diff
@@ -1,5 +1,17 @@
...
 ]
+
+
+# Set minimum timeout for all tests to 30 seconds.
+# If some test needs bigger timeout, please override it for the specific test
+# using @pytest.mark.timeout annotation
+def pytest_collection_modifyitems(items):
+    for item in items:
+        if item.get_closest_marker('timeout') is None:
+            item.add_marker(pytest.mark.timeout(30))
```
Comment (Yarboa) [inline tests/conftest.py]: Please consider change it to 45
Please refer the discussions here 
https://github.com/containers/bluechi/pull/516

**Review by Yarboa — APPROVED** [review]

File: tests/containers/integration-test-packit [hunk 2]
```diff
@@ -0,0 +1,24 @@
+FROM quay.io/bluechi/integration-test-base:latest
```
Comment (engelmi) [inline tests/containers/integration-test-packit]: Couldn't we reuse the `integration-test-snapshot` image and pass in the `$copr_repo` also there? (which would be`@centos-automotive-sig/bluechi-snapshot`)

File: tests/scripts/tests-setup.sh [hunk 3]
```diff
@@ -3,12 +3,32 @@
 
 set -x
 
-if [ "$CONTAINER_USED" = "integration-test-snapshot" ]; then
-   export ARCH="--arch=$(uname -m)"
+# Install required dependencies only when asked
+if [ "$INSTALL_DEPS" == "yes" ]; then
+    dnf install \
```
Comment (engelmi) [inline tests/scripts/tests-setup.sh:8]: Isn't it possible to define those dependencies in the `.packit.yml`? E.g. for `srpm_build_deps`.

Looking at the GH CI, `podman` (python bindings), `pytest` and `pytest-timeout` are installed via pip - I assume the versions installed via `dnf` are better aligned, right?

File: tests/scripts/tests-setup.sh [hunk 4]
```diff
@@ -3,12 +3,32 @@
...
+fi
+
+# When building from packit testing farm we need to pass correct copr repo to the container
+BUILD_ARG=""
+if [ "$PACKIT_COPR_PROJECT" != "" ]; then
+    BUILD_ARG="--build-arg copr_repo=$PACKIT_COPR_PROJECT"
 fi
 
-podman build $ARCH -f ./containers/$CONTAINER_USED -t $BLUECHI_IMAGE_NAME .
+if [ "$CONTAINER_USED" = "integration-test-snapshot" ]; then
```
Comment (engelmi) [inline tests/scripts/tests-setup.sh]: Not really part of this PR, but is this check really needed? Can't we just always detect `arch`?

File: tests/containers/integration-test-packit (hunk 2 above)
Comment (mwperina) [reply tests/containers/integration-test-packit]: For each PR packit creates new temporary COPR repo for unmerged RPMs. For this PR it's https://copr.fedorainfracloud.org/coprs/packit/eclipse-bluechi-bluechi-488/ that's why I need to pass different copr repo for each PR.

Only after merge RPMs will be pushed to `@centos-automotive-sig/bluechi-snapshot`

File: tests/scripts/tests-setup.sh (hunk 3 above)
Comment (mwperina) [reply tests/scripts/tests-setup.sh:8]: So packit use COPR to build RPMs, so `srpm_build_deps` are just packages, which needs to be installed to the plain system to be able to build SRPM. Once we have SRPM, COPR itself will install build dependencies as described in the spec file.

But integration tests are running on a testing farm and packit just uses testing farm API to run tests, but testing farm is independent to packit. Officially all packages required to run tests within tmt framework should be installed as a part of `prepare` phase of each plan using `packages` keyword: https://tmt.readthedocs.io/en/stable/spec/plans.html#prepare

Unfortunately tmt doesn't recognize properly ubuntu, so using `packages` keyword will break current GH CI, because `dnf install` doesn't work on ubuntu, that's why we install those packages in https://github.com/eclipse-bluechi/bluechi/blob/main/.github/workflows/integration-tests.yml#L49
For now I've used dnf only, because it's much easier that pip, but if we really need newer versions not in Fedora we could switch

File: tests/scripts/tests-setup.sh (hunk 4 above)
Comment (mwperina) [reply tests/scripts/tests-setup.sh]: Hmm, good catch, it's just a remain, which is no longer needed after https://github.com/eclipse-bluechi/bluechi/pull/441
I will remove it

File: tests/scripts/tests-setup.sh (hunk 3 above)
Comment (engelmi) [reply tests/scripts/tests-setup.sh:8]: Ah ok, thanks for the explanation! The packit + testing integration is quite complex

File: tests/containers/integration-test-packit (hunk 2 above)
Comment (mwperina) [reply tests/containers/integration-test-packit]: Fixed

**Review by engelmi — APPROVED** [review]
LGTM
